core/src/main/java/org/mule/runtime/core/internal/processor/strategy/StreamPerThreadSink.java

```diff
@@ -43,47 +33,23 @@
 public class StreamPerThreadSink implements Sink, Disposable {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(StreamPerThreadSink.class);
-  private static final int THREAD_CACHE_TIME_LIMIT_IN_MINUTES = 60;
-  private static final int TRANSACTION_CACHE_TIME_LIMIT_IN_MINUTES = 10;
 
-  private final ReactiveProcessor processor;
-  private final Consumer<CoreEvent> eventConsumer;
   private final FlowConstruct flowConstruct;
 
   private volatile boolean disposing = false;
-  private final Cache<Thread, FluxSink<CoreEvent>> sinks =
-      Caffeine.newBuilder().weakKeys().expireAfterAccess(THREAD_CACHE_TIME_LIMIT_IN_MINUTES, MINUTES).build();
-  private final Cache<Transaction, FluxSink<CoreEvent>> sinksNestedTx =
-      Caffeine.newBuilder().weakKeys().expireAfterAccess(TRANSACTION_CACHE_TIME_LIMIT_IN_MINUTES, MINUTES).build();
+
   // We add this counter so we can count the amount of finished sinks when disposing
   // The previous way involved having a strong reference to the thread, which caused MULE-19209
-  private AtomicLong disponsableSinks = new AtomicLong();
-
-  /**
-   * Creates a {@link StreamPerThreadSink}.
-   *
-   * @param processor     the processor to process events emitted onto stream, typically this processor will represent the flow
-   *                      pipeline.
-   * @param eventConsumer event consumer called just before {@link CoreEvent}'s emission.
-   */
-  public StreamPerThreadSink(ReactiveProcessor processor, Consumer<CoreEvent> eventConsumer, FlowConstruct flowConstruct) {
-    this.processor = processor;
-    this.eventConsumer = eventConsumer;
+  private AtomicLong disposableSinks = new AtomicLong();
+  private AbstractCacheSinkProvider sinkProvider;
+
+  public StreamPerThreadSink(FlowConstruct flowConstruct, AbstractCacheSinkProvider sinkProvider) {
```

Comment:
I think the dependency injected here should be an interface. 
Probably the StreamPerThreadThink should be renamed to something like ReactorSinkProviderBasedSink and the one that has the logic corresponding to the cache is the provider.